### DISCUSSION TITLE
FreeSewing v4 wishlist

### DISCUSSION THREAD

**Discussion by joostdecock** [freesewing/freesewing#6947] [Announcements]
NodeJS 20 will reach its end of life on 22 October. We will switch to NodeJS 22 accordingly.

As that's a breaking change, that means FreeSewing v4 will be released this autumn.

Please take a moment to think about any breaking changes you'd like to see as there's still time to consider them for v4 inclusion.
You can leave your comments below.

## v4 wishlist

- [ ] Drop yarn in favor of NPM now that it supports workspaces (suggested by @HaasJona)

**Comment by BenJamesBen** [comment]
I'm not sure if this is a breaking change, but I was thinking of functionality we could introduce with v4:

When commands like `npx @freesewing/new-design` and `yarn lab` are executed, it might be nice to have some sort of "version checking script" be the first thing that gets run.
1. The first thing the script would do would be to print debug info: Node version, npm version, yarn version, nvm version, what platform the script is being run on (Windows, Linux, Mac), the version of the operating system, etc. When people experience problems running FreeSewing, we could simply ask them to copy/paste the debug info to help with problem-solving.
2. If there is a known issue, for example if we know that FreeSewing doesn't currently work with Node 24 and that we recommend that people use Node 20, the script could flag those sorts of issues. Perhaps the script could even prevent FreeSewing from being run (and subsequently crashing) and tell the user what the problem and solution is: The version of Node you are using is too old/new, etc.
